source/Engine.h

```diff
@@ -159,7 +161,13 @@ class Engine {
 
 	void CalculateStep();
 
-	void MoveShip(const std::shared_ptr<Ship> &ship);
+	// Thread-safe wrapper for MoveShip.
+	using ShipResourceProvider = ResourceProvider<std::list<Visual>, std::list<std::shared_ptr<Flotsam>>,
+			std::vector<std::shared_ptr<Ship>>, std::vector<Projectile>>;
+	void MoveShip(const std::shared_ptr<Ship> &ship, ShipResourceProvider &provider);
+	void MoveShip(const std::shared_ptr<Ship> &ship, std::list<Visual> &visuals,
```

Comment:
~~I think it was you who wanted me to synchronize comments between header and source~~